### PR TITLE
Fixed code analysis errors

### PR DESCRIPTION
**Pull request by jerome-jacob**
https://github.com/KhronosGroup/OpenCOLLADA/issues/240

### CODE REVIEW

File: Externals/MayaDataModel/include/MayaDMBlendShape.h [hunk 1]
```diff
@@ -29,31 +29,31 @@ class BlendShape : public GeometryFilter
...
 				}
 			}* inputTargetItem;
+			size_t nbInputTargetItem;
 			float* targetWeights;
+			size_t nbTargetWeights;
 			void write(FILE* file) const
 			{
-				size_t size = sizeof(inputTargetItem)/sizeof(InputTargetItem);
-				for(size_t i=0; i<size; ++i)
+				for (size_t i = 0; i<nbInputTargetItem; ++i)
```
Comment (RemiArnaud) [inline Externals/MayaDataModel/include/MayaDMBlendShape.h]: I'm missing something most likely I don't see where nbInputTargetItem is initialized ? 
The same question goes for all the other code changes in this PR.

Comment (jerome-jacob) [inline Externals/MayaDataModel/include/MayaDMBlendShape.h]: nbInputTargetItem must be initialized by user, just like inputTargetItem pointer. FYI this part of the code is not used by OpenCOLLADA.

**Comment by RemiArnaud** [comment]
@jerome-jacob - please just remove broken and unused functions.
thanks

**Comment by mbarnes-sb** [comment]
Better code quality is a good thing and will contribute towards porting to newer compilers in the coming months/years.

**Comment by RemiArnaud** [comment]
@jerome-jacob - I understand this code will be used now that it's fixed.  Can you please merge with master so we can rebuild and then merge

Thanks!

**Pull request merged**